app/models/taxon.rb

```diff
@@ -37,6 +38,20 @@ def redirected?
     publication_state == "unpublished" && !redirect_to.nil?
   end
 
+  def draft_previously_published?
+    previous_state, latest_state = last_two_publication_states
+    previous_state && latest_state == "draft"
+  end
+
+  def ordered_publication_state_history
+    ordered_history = state_history.sort_by { |k, _v| k }
+    ordered_history.map!(&:second)
```

Comment:
Would this work instead?
```ruby
state_history.sort_by(&:first).map(&:second)
```

This would remove the use of `map!`, and the `ordered_history` variable.

Reply:
I'lI give it a try.